src/Time/TimeSteppers/AdamsBashforthN.cpp

```diff
@@ -0,0 +1,142 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Time/TimeSteppers/AdamsBashforthN.hpp"
+
+#include <algorithm>
+
+#include "Utilities/Gsl.hpp"
+
+namespace TimeSteppers {
+
+AdamsBashforthN::AdamsBashforthN(size_t target_order, bool self_start)
+    : target_order_(target_order), is_self_starting_(self_start) {
+  if (target_order_ < 1 or target_order_ > MAXIMUM_ORDER) {
+    ERROR("The order for Adams-Bashforth Nth order must be 1 <= order <= "
+          << MAXIMUM_ORDER);
+  }
+}
+
+size_t AdamsBashforthN::number_of_substeps() const noexcept {
+  return 1;
+}
+
+size_t AdamsBashforthN::number_of_past_steps() const noexcept {
+  return target_order_ - 1;
+}
+
+bool AdamsBashforthN::is_self_starting() const noexcept {
+  return is_self_starting_;
+}
+
+double AdamsBashforthN::stable_step() const noexcept {
+  if (target_order_ == 1) {
+    return 1.;
+  }
+
+  // This is the condition that the characteristic polynomial of the
+  // recurrence relation defined by the method has the correct sign at
+  // -1.  It is not clear whether this is actually sufficient.
+  const auto& coefficients = gsl::at(coefficients_, target_order_ - 2);
+  double invstep = 0.;
+  double sign = 1.;
+  for (const auto coef : coefficients) {
+    invstep += sign * coef;
+    sign = -sign;
+  }
+  return 1. / invstep;
+}
+
+std::vector<double> AdamsBashforthN::get_coefficients_impl(
+    const std::vector<double>& steps) {
+  const size_t order = steps.size();
+  ASSERT(order >= 1 and order <= MAXIMUM_ORDER, "Bad order" << order);
+  if (order == 1) {
+    return {1.};
+  }
+  if (std::all_of(steps.begin(), steps.end(),
+                  [=](const double& s) { return s == 1.; })) {
+    return gsl::at(coefficients_, order - 2);
+  }
+
+  return variable_coefficients(steps);
+}
+
+std::vector<double> AdamsBashforthN::variable_coefficients(
+    const std::vector<double>& steps) {
+  const size_t order = steps.size();
+
+  // polynomials[coefficient][term exponent]
```

Comment:
Unless this happens to be something general like Jacobi polynomials I think a reference + an equation is sufficient, just so that if someone in the future tries to make improvements on anything by using different formulas they can find the original.